[#116941] Implement the controller action_access_control_list

ventilooo opened the pull request.

Commit:
[#116941] Work In Progress

Commit:
[10.0] implement the controller for module action_access_control_list

joaoalf commented on action_access_control_list/controllers/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License GPL-3.0 or later (http://www.gnu.org/licenses/gpl).
```
@ventilooo LGPL please.

joaoalf commented on action_access_control_list/models/ir_action_access.py:
```diff
@@ -4,64 +4,57 @@
     group_id = fields.Many2one(
-        string='Group',
-        comodel_name='res.groups',
+        'res.groups',
```
@ventilooo Aesthetic changes should be avoided unless it's made to comply with an already established projet style.

dufresnedavid replied:
@joaoalf this is my change. The module is still not in production. This was done only to make the code more standard.

joaoalf reviewed (changes requested):
- Commit msg (WIP) ??
- License mismatch

A PR comment explaining what we're trying to accomplish with these changes would be nice.

Commit:
[10.0] add README to action_access_control_list

Commit:
[10.0] Fix licence

Commit:
[10.0] fix unit tests

Commit:
[10.0] action_access_control_list add web interface introspection

In the web interface, hide any action button that the user is not allowed to click on.

Show to the user all buttons that have a group hardcoded and for which the user has the access to click on.

joaoalf commented on action_access_control_list/controllers/main.py:
```diff
@@ -0,0 +1,92 @@
+
+EXCLUDED_MODELS = {'res.users'}
+
+BASE_METHODS = {
```
@dufresnedavid I'm not sure what's your interpretation of base method but the list seems incomplete to me. What about `load`, `search_read`, `browse`, `read_group`, `name_create`, `fields_view_get`...

dufresnedavid replied:
@joaoalf right, I need to add the missing methods.

joaoalf commented on test_action_access_control_list/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License GPL-3.0 or later (http://www.gnu.org/licenses/gpl).
```
@dufresnedavid GPL -> LGPL

joaoalf commented on action_access_control_list/models/ir_protected_action.py:
```diff
@@ -1,40 +1,109 @@
-    """
+    """Secured Action"""
 
     _name = 'ir.protected.action'
```
@dufresnedavid IMHO `ir.action.protected` would be a more standard name.

dufresnedavid replied:
@joaoalf I agree

joaoalf commented on action_access_control_list/README.rst:
```diff
@@ -2,8 +2,45 @@
+----------------
+Standard object methods (such as read, write, create, unlink, onchange, copy, etc) are managed differently than custom object methods (actions).
+
+If a standard method is called, the access will be checked only for the model being manipulated.
```
@dufresnedavid Looks like you're changing standard behaviour for the base methods and if it's the case, why extrapolate the module objective?

Commit:
Update README

Commit:
[10.0] Fix access domain check when user has multiple groups
